processor/src/main/java/org/mapstruct/ap/internal/model/AdditionalAnnotationsBuilder.java

```diff
@@ -90,6 +92,33 @@ protected void addInstances(AnnotateWithsGem gem, Element source, Set<Annotation
         }
     }
 
+    @Override
+    public Set<Annotation> getProcessedAnnotations(Element source) {
+        Set<Annotation> processedAnnotations = super.getProcessedAnnotations( source );
+        return addDeprecatedAnnotation( source, processedAnnotations );
+    }
+
+    private Set<Annotation> addDeprecatedAnnotation(Element source, Set<Annotation> annotations) {
+        Type deprecatedType = typeFactory.getType( Deprecated.class );
+        Optional<? extends AnnotationMirror> optionalAnnotationMirror = source.getAnnotationMirrors().stream()
+                .filter( annotationMirror -> DEPRECATED_ANNOTATION_FQN.equals(
+                        annotationMirror.getAnnotationType().asElement().toString() ) ).findFirst();
+        if (optionalAnnotationMirror.isPresent() &&
+                annotations.stream().anyMatch( annotation -> annotation.getType().equals( deprecatedType ) ) ) {
+            messager.printMessage(
```

Comment:
I think that for this particular case we don't have to add an error. We can have a warning that is saying that the method is already using `@AnnotateWith(Deprecated.class)` and that doing it twice is obsolete.

Reply:
@filiphr , should the warning take into account the situation where the `@Deprecation` and the `@AnnotateWith` don't match?

for example:
```java
@Deprecated( since = "1.5" )
@AnnotateWith(
    value = Deprecated.class,
    elements = {
        @Element( name = "forRemoval", boolean = true ),
        @Element( name = "since", string = "1.5.0" )
    }
)
```

Reply:
Okay, so when there is a duplicate definition, a warning is raised and the type defined in @AnnotateWith(Deprecated.class) is returned

Reply:
Okay, so when there is a duplicate definition, a warning is raised and the type defined in @AnnotateWith(Deprecated.class) is returned